src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
```

Comment:
> On the other hand if you add markers with data that is added through Batch API it might be assumed that you want those markers synchronized (added to document, undo, etc.) too. I am fine with this.

Yea, I think that adding fragment with markers to the document is... adding markers to the document. One expects that they will be added. If I want to add only elements, I add elements and markers add to the `document.markers` directly, without `batch` API.